### PR TITLE
Strand check cron job

### PR DESCRIPTION
**Pull request by jango2106**
Added cron job for checking strand availability. Has a application property to adjust duration

## Proposed Changes

  - Added a cron job that runs every 10 mins (configurable) to check for strand availability

### CODE REVIEW

File: server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java [hunk 1]
```diff
@@ -22,7 +32,43 @@ public StrandService(StrandRepository strandRepository) {
...
+              logger.error(
+                  "Strand " + strand.getName() + " failed the health check with a non-200 status.");
+              strand.setAvailable(false);
+            }
+          } catch (RestClientException ex) {
+            logger.error(
+                "Strand "
+                    + strand.getName()
+                    + " failed the health check with the following error: "
+                    + ex.getMessage());
```
Comment (ahoopes16) [inline server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java]: Oh man this makes me miss formatted strings in JS and Python

File: server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java [hunk 2]
```diff
@@ -22,7 +32,43 @@ public StrandService(StrandRepository strandRepository) {
...
-        new StrandModel(request.getName(), request.getIpAddress(), request.getPort());
+        StrandModel.builder()
+            .name(request.getName())
+            .ipAddress(request.getIpAddress())
+            .port(request.getPort())
+            .build();
     return strandRepository.save(strand);
   }
+
+  // Run strands check every 10 minutes
```
Comment (ahoopes16) [inline server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java]: I'd be wary of commenting implementation specific things like this - they turn into lies very easily, especially whenever this thing ideally becomes an environment variable

File: server/src/main/resources/application.yml [hunk 3]
```diff
@@ -5,3 +5,7 @@ spring:
       host: "http://localhost/"
       port: 27017
       uri: "mongodb://localhost/paracord"
+
+schedule:
+  strandAvailabilityMS: 1000
```
Comment (ahoopes16) [inline server/src/main/resources/application.yml]: Is this 10 minutes? What unit of time does the scheduler use?

File: server/src/test/java/com/showmetheroapes/paracord/services/StrandServiceTest.java [hunk 4]
```diff
@@ -3,22 +3,34 @@
...
+@AutoConfigureWebClient(registerRestTemplate = true)
+@RestClientTest(components = {StrandService.class, StrandRepository.class})
 class StrandServiceTest {
-  @Mock StrandRepository strandRepository;
+  @MockBean StrandRepository strandRepository;
 
-  @InjectMocks StrandService strandService;
+  @Autowired StrandService strandService;
+
+  @Autowired MockRestServiceServer otherServer;
```
Comment (ahoopes16) [inline server/src/test/java/com/showmetheroapes/paracord/services/StrandServiceTest.java]: Well that's super handy lol

File: server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java [hunk 5]
```diff
@@ -4,16 +4,26 @@
...
+  private static final Logger logger = LoggerFactory.getLogger(StrandService.class);
+
   private StrandRepository strandRepository;
+  private RestTemplate restTemplate;
 
   @Autowired
-  public StrandService(StrandRepository strandRepository) {
+  public StrandService(StrandRepository strandRepository, RestTemplate restTemplate) {
     this.strandRepository = strandRepository;
+    this.restTemplate = restTemplate;
```
Comment (ahoopes16) [inline server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java]: So I'm confused how this rest template is connected to the rest template in the `CronJobConfiguration`.. please help meeeeee

File: server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java (hunk 1 above)
Comment (jango2106) [reply server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java]: Tell me about it

File: server/src/main/resources/application.yml (hunk 3 above)
Comment (jango2106) [reply server/src/main/resources/application.yml]: Should have been milliseconds... makes your comment about comments really come to light lol

Comment (jango2106) [reply server/src/main/resources/application.yml]: And also 600000 should be the value

File: server/src/test/java/com/showmetheroapes/paracord/services/StrandServiceTest.java (hunk 4 above)
Comment (jango2106) [reply server/src/test/java/com/showmetheroapes/paracord/services/StrandServiceTest.java]: Agreed!

File: server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java (hunk 5 above)
Comment (jango2106) [reply server/src/main/java/com/showmetheroapes/paracord/services/StrandService.java]: This would be easier to explain in person

**Pull request merged**